### PR TITLE
Upgrade ember 5

### PR DESCRIPTION
**Pull request by C0rydoras**
Tests will fail due to mirage not being compatible with ember-data 5 yet (https://github.com/miragejs/ember-cli-mirage/pull/2444)

# This PR includes
- set `async` and `inverse` on models
- removed usage of various deprecated methods from project-form
- upgraded ember-{source,cli} to 4.12 (`npx ember-cli-update --to 4.12`)
- changed location type to historic (auto was deprecated)
- import and return components (`power-select`, `power-select-multiple`) instead of using `ensureSafeComponent` with their name
- added `useSessionSetupMethod` to `ember-cli-build.js`
- upgraded `ember-data` to 4.12 which required some [extra work](https://github.com/embroider-build/embroider/issues/1322)
- upgraded to ember-5 by running `npx ember-cli-update --to 5.0`

### CODE REVIEW

File: ember/.eslintrc.js [hunk 1]
```diff
@@ -4,5 +4,52 @@ module.exports = {
...
+        node: true,
+      },
+      extends: ['plugin:n/recommended'],
+    },
+    {
+      // test files
+      files: ['tests/**/*-test.{js,ts}'],
+      extends: ['plugin:qunit/recommended'],
+    },
+  ],
```
Comment (anehx) [inline ember/.eslintrc.js:54]: All of this is included in `@adfinis/eslint-config/ember-app`. You can revert those changes

**Review by anehx — CHANGES_REQUESTED** [review]
Small nitpick in the eslint config, otherwise good job!

File: ember/.eslintrc.js (hunk 1 above)
Comment (C0rydoras) [reply ember/.eslintrc.js]: done here: https://github.com/adfinis/Outdated/pull/37/commits/8fa3d74765a720ca5602616a58a4fb9c211a981b

**Comment by anehx** [comment]
LGTM, but your tests are failing.

**Comment by C0rydoras** [comment]
@anehx The tests are failing due to mirage (https://github.com/miragejs/ember-cli-mirage/pull/2444)

**Review by anehx — APPROVED** [review]

**Pull request merged**